pkg/core/native/native_gas.go

```diff
@@ -80,15 +81,20 @@ func (g *GAS) Initialize(ic *interop.Context) error {
 
 // OnPersist implements Contract interface.
 func (g *GAS) OnPersist(ic *interop.Context) error {
-	//for _ ,tx := range ic.block.Transactions {
-	//	g.burn(ic, tx.Sender, tx.SystemFee + tx.NetworkFee)
-	//}
-	//validators := g.NEO.getNextBlockValidators(ic)
-	//var netFee util.Fixed8
-	//for _, tx := range ic.block.Transactions {
-	//	netFee += tx.NetworkFee
-	//}
-	//g.mint(ic, <primary>, netFee)
+	for _, tx := range ic.Block.Transactions {
+		absAmount := big.NewInt(int64(tx.SystemFee.FloatValue() + tx.NetworkFee.FloatValue()))
```

Comment:
Never. Never. NEVER calculate anything money-like in floats. Just sum them, no need to cast to floats.

Reply:
What about decimals in this case?

Reply:
Everywhere GAS is being calculated or stored internally it's in some kind of Fixed8 form, that is where 1 is actually 0.00000001 and it works fine. You only deal with 0.some GAS in UI.